core/variant/variant.cpp

```diff
@@ -831,6 +831,16 @@ bool Variant::can_convert_strict(Variant::Type p_type_from, Variant::Type p_type
 	return false;
 }
 
+bool Variant::conversion_causes_copy(const Variant::Type &p_type_from, const Variant::Type &p_type_to) {
+	if (p_type_from == p_type_to) {
+		return false;
+	}
+	if (p_type_to == NIL || p_type_to == NIL) {
+		return false;
+	}
+	return (Variant::is_type_shared(p_type_from) || p_type_from >= PACKED_BYTE_ARRAY) && (Variant::is_type_shared(p_type_to) || p_type_to >= PACKED_BYTE_ARRAY);
```

Comment:
`is_type_shared` is `false` for the packed arrays - is this correct?

Reply:
Yes, because they're copied across the C++/scripting boundaries. They are only passed by reference within GDScript.

Reply:
This should be implemented in GDScript, actually. It is a language-specific thing, so it does not need to be available in Variant.